core/block/import/objectid/provider.go

```diff
@@ -0,0 +1,60 @@
+package objectid
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/anyproto/any-sync/commonspace/object/tree/treestorage"
+
+	"github.com/anyproto/anytype-heart/core/block/import/converter"
+	"github.com/anyproto/anytype-heart/core/block/object/objectcache"
+	"github.com/anyproto/anytype-heart/pkg/lib/core"
+	sb "github.com/anyproto/anytype-heart/pkg/lib/core/smartblock"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/objectstore"
+)
+
+type IDProvider interface {
+	GetID(spaceID string, sn *converter.Snapshot, createdTime time.Time, getExisting bool) (string, treestorage.TreeStorageCreatePayload, error)
```

Comment:
Maybe name it `GetIdAndPayload` or something like this? I think the mention of payload is important here

Reply:
Done